lib/awspec/type/alb.rb

```diff
@@ -0,0 +1,15 @@
+module Awspec::Type
+  class Alb < Base
+    def resource_via_client
+      @resource_via_client ||= find_alb(@display_name)
+    end
+
+    def id
+      @id = resource_via_client.load_balancer_name if resource_via_client
+    end
+
+    def dns_name
```

Comment:
When I remove `resource_via_client` I get an exception. This method is actually defined in the base class, so `method_missing` doesn't get called.

```
  1) alb 'my-alb' should exist
     Failure/Error: raise 'this method must be override!'

     RuntimeError:
       this method must be override!
     # ./lib/awspec/type/base.rb:35:in `resource_via_client'
     # ./lib/awspec/type/alb.rb:8:in `id'
     # ./lib/awspec/type/base.rb:27:in `exists?'
     # ./spec/type/alb_spec.rb:5:in `block (2 levels) in <top (required)>'

```